README.md

```diff
@@ -97,45 +95,91 @@ _Only when accessing a property of this module, would the execution operations b
+[source phase imports](https://github.com/tc39/proposal-source-phase-imports) proposal:
 
 ```js
-// using import attributes (`lazyInit` to illustrated, but it could also be `deferEval`, say)
-import lazyModuleNamespace from "y" with { lazyInit: true }
-
-// with an explicit "*" to indiciate the namespace form
-import * as lazyModuleNamespace from "y" with { lazyInit: true }
-
 // or with a custom keyword:
-lazy import * as yNs from "y";
+import defer * as yNamespace from "y";
 ```
 
-There are other proposals, but the discussion of the naming may derail the discussion of the
-semantics, which should be agreed on first and would likely guide the design of the API.
-
 ## Semantics
 
-The semantics would be similar to import reflection, in that the deferred module namespace can be
-thought of as a leaf node that does not add further branches to the execution graph.
-
 The imports would still participate in deep graph loading so that they are fully populated into
-the module cache prior to execution.
+the module cache prior to execution, however it the imported module will not be evaluated yet.
 
-When a property of a `DeferredModuleNamespace` is accessed, if the execution has not already
-been performed, a new top-level execution would be initiated for that module.
+When a property of the resulting module namespace object is accessed, if the execution has
+not already been performed, a new top-level execution would be initiated for that module.
 
 In this way, a deferred module evaluation import acts as a new top-level execution node
 in the execution graph, just like a dynamic import does, except executing synchronously.
 
-If accessing a graph subject to top-level await, an error would be thrown. Eager execution of the
-asynchronous components of top-level await graphs could be a possible option for enabling these
-use cases as well.
+### Top-level await
+
+Property access on the namespace object of a deferred module must be synchronous, and it's thus
+impossible to defer evaluation of modules that use top-level await. When a module is imported
+using the `import defer` syntax, its asynchronous dependencies together with their own transitive
+dependencies are eagerly evaluated, and only the synchronous parts of the graph are deferred.
+
+Consider the following example, where `a` is the top-level entry point:
+<table><tr><td>
+
+```js
+// a
+import "b";
+import defer * as c from "c"
+
+setTimeout(() => {
+  c.value
+}, 1000);
+```
+
+</td><td>
+
+```js
+// b
+```
+
+</td><td>
+
+```js
+// c
+import "d"
+import "f"
+export let value = 2;
+```
+
+</td></tr><tr><td>
+
+```js
+// d
+import "e"
+await 0;
+```
+
+</td><td>
+
+```js
+// e
+```
+
+</td><td>
+
+```js
+// f
+```
+
+</td></tr></table>
+
+Since `d` uses top-level await, `d` and its dependencies cannot be deferred. The initial evaluation will execute `b`, `e`, `d` and `a`. Later, the `c.value` access will trigger the execution of `f` and `c`.
```

Comment:
Perhaps further explain here what is eagerly evaluated at load time?

Reply:
Ah ok that seems fine. Separating out the steps as bullet points might help follow the algorithm better but you've certainly captured it.